core/src/main/scala/akka/kafka/internal/SubSourceLogic.scala

```diff
@@ -78,28 +80,52 @@ private[kafka] abstract class SubSourceLogic[K, V, Msg](
   private implicit val askTimeout = Timeout(10000, TimeUnit.MILLISECONDS)
   def partitionAssignedCB(tps: Set[TopicPartition]) = {
     implicit val ec = materializer.executionContext
-    getOffsetsOnAssign.fold(pumpCB.invoke(tps)) { getOffsets =>
-      getOffsets(tps)
+
+    val partitions = tps -- partitionsToRevoke
+
+    if (partitions.nonEmpty) {
+      log.debug(s"Assigning new partitions: ${partitions.mkString(", ")}")
```

Comment:
For logging you should use the `log.debug("message {}", value)` notation as it defers string creation.
In the particular case you might just add `log.isDebugEnabled` to the if clause.